drake/systems/frames/@CoordinateFrame/CoordinateFrame.m

```diff
@@ -68,17 +57,26 @@
         ind=ind+1;
       end
       if (nargin<4 || isempty(coordinates))
-        obj.coordinates=cellfun(@coordinateName,cell(dim,1),'UniformOutput',false);
+        coordinates=cellfun(@coordinateName,cell(dim,1),'UniformOutput',false);
       else
         typecheck(coordinates,'cell');
         sizecheck(coordinates,dim);
         for i=1:dim
           typecheck(coordinates{i},'char');
         end
-        obj.coordinates = {coordinates{:}}';
+        coordinates = {coordinates{:}}';
       end
+      
```

Comment:
This assumes that the derived class has set up its `mex_ptr` before calling the superclass constructor, right?